metrics_api/lib/opentelemetry/metrics/meter.rb

```diff
@@ -8,71 +8,156 @@ module OpenTelemetry
+      # @param name [String]
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param callback [optional Proc, Array<Proc>]
+      #   Callback functions should:
+      #   - be reentrant safe;
+      #   - not take an indefinite amount of time;
+      #   - not make duplicate observations (more than one Measurement with the same attributes)
+      #     across all registered callbacks;
+      #
+      # @return [Instrument::ObservableGauge]
+      def create_observable_gauge(name, unit: nil, description: nil, callback: nil)
+        create_instrument(:observable_gauge, name, unit, description, callback) { NOOP_OBSERVABLE_GAUGE }
       end
 
-      def create_observable_up_down_counter(name, unit: nil, description: nil, callback:)
-        create_instrument(:observable_up_down_counter, name, unit, description, callback) { OBSERVABLE_UP_DOWN_COUNTER }
+      # @param name [String]
+      #   Must conform to the instrument name syntax:
+      #   not nil or empty, case-insensitive ASCII string that matches `/\A[a-z][a-z0-9_.-]{0,62}\z/i`
+      # @param unit [optional String]
+      #   Must conform to the instrument unit rule:
+      #   case-sensitive ASCII string with maximum length of 63 characters
+      # @param description [optional String]
+      #   Must conform to the instrument description rule:
+      #   UTF-8 string but up to 3 bytes per charater with maximum length of 1023 characters
+      # @param callback [optional Proc, Array<Proc>]
+      #   Callback functions should:
+      #   - be reentrant safe;
+      #   - not take an indefinite amount of time;
+      #   - not make duplicate observations (more than one Measurement with the same attributes)
+      #     across all registered callbacks;
+      #
+      # @return [Instrument::ObservableUpDownCounter]
+      def create_observable_up_down_counter(name, unit: nil, description: nil, callback: nil)
+        create_instrument(:observable_up_down_counter, name, unit, description, callback) { NOOP_OBSERVABLE_UP_DOWN_COUNTER }
       end
 
       private
 
       def create_instrument(kind, name, unit, description, callback)
-        raise InstrumentNameError if name.nil?
-        raise InstrumentNameError if name.empty?
-        raise InstrumentNameError unless NAME_REGEX.match?(name)
-        raise InstrumentUnitError if unit && (!unit.ascii_only? || unit.size > 63)
-        raise InstrumentDescriptionError if description && (description.size > 1023 || !utf8mb3_encoding?(description.dup))
```

Comment:
After reading [Synchronous Instrument API](https://opentelemetry.io/docs/reference/specification/metrics/api/#synchronous-instrument-api) I noticed the following

> The API SHOULD be documented in a way to communicate to users that the
> `name` parameter needs to conform to the instrument name syntax.
> **The API SHOULD NOT validate the `name`; that is left to
> implementations of the API.**

and

> The `unit` parameter needs to support the instrument unit rule.
> Meaning, the API MUST accept a case-sensitive string that supports
> ASCII character encoding and can hold at least 63 characters.
> **The API SHOULD NOT validate the `unit`.**

Similar for `name` and `unit` in [Asynchronous Instrument API](https://opentelemetry.io/docs/reference/specification/metrics/api/#asynchronous-instrument-api)

So, I think we should remove these validations from here and add yard doc comments instead. What do you think?

Reply:
>So, I think we should remove these validations from here and add yard doc comments instead. What do you think?

I think that's reasonable, the API spec is explicit in that it should perform any validations against the name field, so that will be the responsibility of the SDK implementation. Yard docs seem like a good solution for this.

Reply:
Also in the "Metrics No-Op API Implementation" docs it makes clear No-Op Meters must not validate. For example:

> The Meter MUST accept these parameters. However, the Meter MUST NOT validate any argument it receives.

https://opentelemetry.io/docs/specs/otel/metrics/noop/#counter-creation